hadoop-ozone/common/src/main/java/org/apache/hadoop/ozone/om/protocolPB/GrpcOmTransport.java

```diff
@@ -93,6 +104,22 @@ public void start() {
             .usePlaintext()
             .maxInboundMessageSize(maxSize);
 
+    if (secConfig.isGrpcTlsEnabled()) {
+      try {
+        SslContextBuilder sslContextBuilder = GrpcSslContexts.forClient();
+        if (caCerts != null) {
+          sslContextBuilder.trustManager(caCerts);
+        } else {
+          LOG.error("x509Certicates empty");
```

Comment:
If TLS is enabled and caCerts is empty, wouldn't authentication fail?

Reply:
Thanks @hanishakoneru for looking at this and for your comments.  
If `grpcTlsEnabled` and the caCerts is empty, then the channel connection between the s3g client (`GrpcOmTransport`) and the om will not be authenticated by the client.  In this case, the error will be logged and the connection will fail.  The client receives a 500 http response "An error occurred (500) (reached max retries)". 

The logged error as a result of TLS enabled with empty caCerts logs that the `"x509Certicates empty"`.  That and the client request failure should provide /alert the necessary info for the admin.

Reply:
So if the client request cannot be authenticated, would it not be better to fail the client request here itself instead of when it reaches the max retries?

Reply:
Errors on s3g client creation appear the same to the user I'm afraid.  The s3 gateway uses jax context and dependency injections that fail if the s3 gateway fails on initialization.  That is what happens when the s3g TLS channel is, for any given reason, unable to be created when _ozone.security.enabled_ and _hdds.grpc.tls.enabled_ are set.  The error in s3g client creation is propagated up through the jax injection failure which results in retries and the 500 http erro response.

 > would it not be better to fail the client request here itself instead of when it reaches the max retries?

In this case all errors that occur during the TLS / Grpc channel creation appear the same to the client with a 500 http response and max retires.  So if we fail it another way, it will still propagate back the same.  We do provide additional valuable information on the error through the log.  Here we have the additional information that we log indicating that the `"x509Certificates empty"`.